### PR TITLE
#15 Provide tagged string template functions

### LINKED ISSUE #15: Provide tagged string template functions
Would be nice to provide tagged string template functions.

Something like:

```js
normalizeWhiteSpaces`
  Hi, ${student.name}!
  How are you doing?
`;
//=> "Hi, Vitor! How are you doing?"
```

Seems that wouldn't be hard to do with current normalize functions. But maybe they should be scoped in a different way.

Some implementation details in the gist below:
https://gist.github.com/VitorLuizC/f80a9b51c3a686e5b5e35259bbf263fa

### PR DESCRIPTION
**Pull request by phuvinhbmt**
Resolves #15

### CODE REVIEW

File: src/normalizeWhiteSpaces.spec.ts [hunk 1]
```diff
@@ -12,4 +12,11 @@ describe('normalizeWhiteSpaces', () => {
       'Hi, how is everything ?'
     );
   });
+
+  it('Normalize white-space in string template', () => {
+    const student = { name: 'John   Doe' };
+    expect(
+      normalizeWhiteSpaces(` Hi,  ${student.name}!\n How are you doing? `)
```
Comment (VitorLuizC) [inline src/normalizeWhiteSpaces.spec.ts:19]: @phuvinhbmt I also used the wrong term in the issue, its [tagged templates](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#tagged_templates), and they're little bit different than just supporting string templates. Tagged templates are functions that receives a list os strings as first argument, and the interpolations as the rest of arguments.

```ts
function normalizeWhiteSpaces(parts: TemplateStringArray, ...values: unknown[]): string;
```

```suggestion
  it('supports usage as tagged template', () => {
    const student = { name: 'John   Doe' };
    expect(
      normalizeWhiteSpaces` Hi,  ${student.name}!\n How are you doing? `
```

**Review by VitorLuizC — CHANGES_REQUESTED** [review]

**Comment by phuvinhbmt** [comment]
@VitorLuizC I updated `normalizeWhiteSpaces` per your comment. Let me know if it needs updating anything. Thanks

**Comment by phuvinhbmt** [comment]
@VitorLuizC do you mind having a look at this pr when you have time please ? Cheers

**Review by VitorLuizC — APPROVED** [review]
LGTM

**Comment by VitorLuizC** [comment]
@phuvinhbmt sorry to take so long, I really have to improve my priorization skills in order to be a good maintainer again. Anyway, your PR was merged, and it was published under version [2.5.0](https://github.com/VitorLuizC/normalize-text/releases/tag/2.5.0).

Thank you a lot!

**Pull request merged**